Dev: Check for merge commits fails on PR for release code

See #887
This check is not required for PRs to merge develop to master.

```
Run NexusPHP/no-merge-commits@v2.0.0
[NOTICE] Collecting token from input...
[INFO] Token collected.
[NOTICE] Instantiating an Octokit client using token...
[INFO] Octokit client is ready.
[NOTICE] Retrieving commits of PR #887...
[INFO] PR #887 contains 30 commits.
[NOTICE] Inspecting commit SHA: 4dcee8f
Error: [ERROR] Commit SHA 4dcee8f is a merge commit!
```
https://github.com/codeigniter4/shield/actions/runs/6452678862/job/17515035014?pr=887

--- a/.github/workflows/no-merge-commits.yml
+++ b/.github/workflows/no-merge-commits.yml
@@ -2,6 +2,8 @@ name: Detect Merge Commits
 
 on:
   pull_request:
+    branches:
+      - develop
 
 permissions:
   contents: read

